### PR TITLE
Refactor ingestion test cases

### PR DESCRIPTION
**Pull request by zhangchiqing**
Refactor the ingestion engine tests.

### CODE REVIEW

File: model/flow/chunk_test.go [hunk 1]
```diff
@@ -104,3 +106,52 @@ func TestChunkList_Indices(t *testing.T) {
 		require.Contains(t, indices, uint64(0), uint64(2), uint64(4))
 	})
 }
+
+func TestChunkIndexIsSet(t *testing.T) {
```
Comment (zhangchiqing) [inline model/flow/chunk_test.go:110]: These tests, which moved from the ingestion engine tests, should belong to this file

File: engine/execution/ingestion/engine_test.go [hunk 2]
```diff
@@ -895,559 +463,330 @@ func logBlocks(blocks map[string]*entity.ExecutableBlock) {
 	}
 }
 
-func TestExecuteBlockInOrder(t *testing.T) {
+// verify that when blocks above the stop height are finalized, they won't
+// be executed
+func TestStopAtHeightWhenFinalizedBeforeExecuted(t *testing.T) {
```
Comment (zhangchiqing) [inline engine/execution/ingestion/engine_test.go:468]: This test replaced the `TestStopAtHeight` test @janezpodhostnik

File: engine/execution/ingestion/engine_test.go [hunk 3]
```diff
@@ -895,559 +463,330 @@ func logBlocks(blocks map[string]*entity.ExecutableBlock) {
...
+		store.AssertNotExecuted(t, "C", blockC.ID())
+		store.AssertNotExecuted(t, "D", blockD.ID())
 	})
 }
 
-// TestStopAtHeightRaceFinalization test a possible race condition which happens
-// when block at stop height N is finalized while N-1 is being executed.
-// If execution finishes exactly between finalization checking execution state and
-// setting block ID to crash, it's possible to miss and never actually stop the EN
-func TestStopAtHeightRaceFinalization(t *testing.T) {
```
Comment (zhangchiqing) [inline engine/execution/ingestion/engine_test.go:1122]: This test is replaced by `TestStopAtHeightWhenExecutionFinalization`

File: engine/execution/ingestion/engine_test.go [hunk 4]
```diff
@@ -895,559 +463,330 @@ func logBlocks(blocks map[string]*entity.ExecutableBlock) {
...
-			}),
-			mock.Anything)
+		// since stop height is C, verify that only A and B are executed, C and D are not executed
+		store.AssertExecuted(t, "A", blockA.ID())
+		store.AssertNotExecuted(t, "B", blockB.ID())
+		store.AssertNotExecuted(t, "C", blockC.ID())
 	})
 }
 
-func TestExecutionGenerationResultsAreChained(t *testing.T) {
```
Comment (zhangchiqing) [inline engine/execution/ingestion/engine_test.go:1229]: This test has nothing to do with ingestion engine logic, removed

**Review by janezpodhostnik — APPROVED** [review]
Very nice cleanup

File: engine/execution/ingestion/mocks/fetcher.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+		byID: make(map[flow.Identifier]struct{}),
+	}
+}
+
+func (r *MockFetcher) FetchCollection(blockID flow.Identifier, height uint64, guarantee *flow.CollectionGuarantee) error {
+	r.byID[guarantee.ID()] = struct{}{}
+	return nil
+}
+
+func (r *MockFetcher) Force() {
```
Comment (sideninja) [inline engine/execution/ingestion/mocks/fetcher.go:20]: why this empty method?

File: model/flow/chunk_test.go [hunk 6]
```diff
@@ -1,8 +1,10 @@
 package flow_test
 
 import (
+	"math/rand"
```
Comment (sideninja) [inline model/flow/chunk_test.go:4]: prefer using crypto/rand as math/rand is deprecated.

**Review by sideninja — APPROVED** [review]

File: engine/execution/ingestion/mocks/fetcher.go (hunk 5 above)
Comment (zhangchiqing) [reply engine/execution/ingestion/mocks/fetcher.go:20]: The real implementation will force the requests to be sent by calling `Force`. In mocks, we don't care if requests are forced to be sent, we only care the collections are actually requested. That's why I left it empty

**Pull request merged**